scripts/test/Muon/elemental_analysis_2/EA_grouping_tab/elemental_analysis_grouping_table_presenter_test.py

```diff
@@ -0,0 +1,151 @@
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_that_table_has_six_columns_when_initialized(self):
+        self.assertEqual(self.view.num_cols(), 6)
+
+    def test_that_model_is_initialized_as_empty(self):
+        self.assert_model_empty()
+
+    def test_that_view_is_initialized_as_empty(self):
+        self.assert_view_empty()
+
+    def test_add_group_to_view(self):
+        self.test_setup()
+        self.presenter.add_group_to_view(self.context.group_context._groups[0], True)
+        self.assertEqual(self.view.num_rows(), 1)
+        self.assertEqual(len(self.model.groups), 3)
+
+    def test_update_model_from_view(self):
+        self.test_setup()
+        for group in self.context.group_context._groups:
+            self.presenter.add_group_to_view(group, True)
+        self.assertEqual(self.view.num_rows(), 3)
+        self.assertEqual(len(self.model.groups), 3)
+
+        self.view.remove_last_row()
+        self.assertEqual(self.view.num_rows(), 2)
+        self.assertEqual(len(self.model.groups), 3)
+
+        self.presenter.update_model_from_view()
+        self.assertEqual(self.view.num_rows(), 2)
+        self.assertEqual(len(self.model.groups), 2)
+
+    def test_update_view_from_model(self):
+        self.test_setup()
+
+        self.presenter.add_group_to_view(self.context.group_context._groups[0], True)
+        self.assertEqual(self.view.num_rows(), 1)
+        self.assertEqual(len(self.model.groups), 3)
+
+        self.presenter.update_view_from_model()
+        self.assertEqual(self.view.num_rows(), 3)
+        self.assertEqual(len(self.model.groups), 3)
+
+    def test_plot_default_case(self):
+        self.test_setup()
+        self.presenter.add_group_to_view(self.context.group_context._groups[0], False)
+        self.assertEqual(self.view.num_rows(), 1)
+        self.assertEqual(len(self.model.groups), 3)
+        analyse_checkbox = self.view.get_table_item(0, 3)
+        self.assertEqual(analyse_checkbox.checkState(), 0)
+
+        self.presenter.plot_default_case()
+        self.assertEqual(analyse_checkbox.checkState(), 2)
+
+    def test_remove_selected_rows_in_view_and_model(self):
+        self.test_setup()
+        for group in self.context.group_context._groups:
+            self.presenter.add_group_to_view(group, True)
+        self.assertEqual(self.view.num_rows(), 3)
+        self.assertEqual(len(self.model.groups), 3)
+
+        groups_remove = ['9999; Detector 1', '9999; Detector 3']
+        self.view._get_selected_row_indices = mock.Mock(return_value=[0, 2])
+        self.presenter.remove_selected_rows_in_view_and_model(groups_remove)
+        self.assertEqual(self.view.num_rows(), 1)
+        self.assertEqual(len(self.model.groups), 1)
```

Comment:
lets check that detector 2 is left as expected. This would pass if it just removed the last two detectors.